File: src/main/java/com/james/Hello.java
package com.james;

public class Hello {
    public static void main(String[] args) {
//        System.out.println("Hello World");
//        new Person().hello();
        Person p = new Person("James", 123f, 456f);
        p.hello();
        p.weight = 65.5f;
        p.height = 185f;
        System.out.println(p.bmi());

        int score = 88;
        System.out.println(score > 80 || score < 90);

        char A = 'A';
        System.out.println(A == 65);

//        int age = 12;
//        Integer age2 = 20;
//        char c = '我';
//        Character c2 = 'A';
//        byte b = 120;
//        float weight = 120;
//        float height = 520;
//        boolean adult = true;
//        String name = "James";

    }
}

File: src/main/kotlin/com/james/Hello.kt
package com.james

fun main(args: Array<String>) {
    val student =Student("James",22,33)
    student.grading()

    var s: String? = null;//加問號可處理null問題
//    println(s!!.length);//驚嘆號
    println(s?.get(3))
    println(s?.substring(3))
//    println("Hello kotlin")
//    Human().hello()
//    val hh = Human(66.5f, 1.5f).hello()
//    val h = Human(66.6f, 1.6f)
    val h = Human(weight = 66.6f, height = 1.6f)//初始值可指定參數給值
    println(h.bmi())

    val score = 88;
    println(score < 80)

    val c: Char = 'A'
    println(c.toInt() == 65)//字元判斷須加上toInt

//    h.hello()
//
//    var age: Int = 20;
//    age = 77;
//
//    var weight: Float = 22.3F
//    val Name: String
//    Name = "James"

}

//class Human(var weight: Float, var height: Float, var neme: String = "James") {//建構初始值  放到最後面
class Human(var neme: String = "James", var weight: Float, var height: Float) {//建構初始值
init {
    println("Testtttt $weight")
}

//    constructor(name:String,weight: Float,height: Float):this(weight,height)


    fun bmi(): Float {
        var bmi = weight / (height * height)
        return bmi;
    }

    fun hello() {
        println("hello kotlin")
    }

}